### ISSUE TITLE
ol.proj.toLonLat only works with projection as a string

### ISSUE THREAD

**Issue by bartvde** [openlayers/openlayers#6938]
this is with the ol package in version 4.2.0

```
var center = proj.toLonLat(view.getCenter(), view.getProjection().getCode());
```

works ok

but

```
var center = proj.toLonLat(view.getCenter(), view.getProjection());
```
does not even though the toLonLat function specifies it can take a ProjectionLike

**Comment by bartvde** [comment]
this check

```
 */
_ol_proj_.getTransform = function(source, destination) {
  var sourceProjection = _ol_proj_.get(source);
```
results in sourceProjection being null

**Comment by bartvde** [comment]
![selection_305](https://user-images.githubusercontent.com/319678/27289778-7415fb54-550b-11e7-9de9-8c3b84e79f06.png)

the instanceof checks fails

**Comment by bartvde** [comment]
ah seems to be a duplicate of #6926 I will check with the latest

**Comment by bartvde** [comment]
ah I actually I have the exact same thing as #6926 in where ol-mapbox-style included ol 4.1.1 and ol 4.2.0 is included in my project's package.json
